modules/services/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/SharingClient.kt

```diff
@@ -228,8 +277,8 @@ class SharingClient(
     private fun Intent.setExtraStream(file: File) = putExtra(EXTRA_STREAM, FileUtil.createUriWithReadPermissions(context, file, this))
 
     interface Listener {
-        fun onShare(request: SharingRequest) = Unit
-        fun onShared(request: SharingRequest, response: SharingResponse) = Unit
+        fun onShare(request: SharingRequest, event: AnalyticsEvent, eventProperties: Map<String, Any>) = Unit
+        fun onShared(request: SharingRequest, response: SharingResponse, event: AnalyticsEvent, eventProperties: Map<String, Any>) = Unit
```

Comment:
`Listener` should be aware of things like analytics. It is a general purpose interface and things like analytics can be its implementation details.